test/test_jit.py

```diff
@@ -1865,6 +1865,43 @@ def forward(self, x):
         res = get_forward(m._c)(x)
         self.assertEqual(res, ref_res)
 
+    def test_replicate_dequantize_in_block(self):
+        class M(torch.nn.Module):
+            def __init__(self, cond):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 1).float()
+
+                self.cond = cond
+
+            def forward(self, x):
+                x = torch.dequantize(x)
+                if self.cond:
+                    x = self.conv(x)
+                else:
+                    x = x + 3
+                return x
+
+        x = torch.randn([1, 3, 10, 10], dtype=torch.float)
+        x = torch.quantize_per_tensor(x, 0.5, 1, torch.quint8)
+        m = torch.jit.script(M(True))
+        ref_res = m(x)
+        FileCheck().check_count("aten::dequantize", 1, exactly=True) \
+                   .run(m.graph)
+        torch._C._jit_pass_replicate_dequantize(m.graph)
+        FileCheck().check_count("aten::dequantize", 2, exactly=True) \
+                   .run(m.graph)
+        # check dequantize is right before CallMethod of conv
+        FileCheck().check("aten::dequantize") \
+                   .check_next("CallMethod") \
+                   .run(m.graph)
+        # check dequantize is right before add
+        FileCheck().check("aten::dequantize") \
+                   .check("aten::dequantize") \
+                   .check_next("aten::add") \
```

Comment:
How does the graph have add? From what I can see above, you are setting `cond` to True so it should have a conv.

Reply:
both blocks of prim::If will be quantized